action/protocol/staking/protocol.go

```diff
@@ -472,6 +473,25 @@ func (p *Protocol) Validate(ctx context.Context, act action.Action, sr protocol.
 	return nil
 }
 
+func (p *Protocol) isActiveCandidate(ctx context.Context, csr CandidateStateReader, cand *Candidate) (bool, error) {
+	featureCtx := protocol.MustGetFeatureCtx(ctx)
+	// before delegate endorsement enabled, candidate is always active unless it's self-stake bucket is unstaked
+	if featureCtx.DisableDelegateEndorsement {
+		if cand.SelfStake.Cmp(p.config.RegistrationConsts.MinSelfStake) < 0 {
+			return false, nil
+		}
+		return true, nil
+	}
+
+	srHeight, err := csr.SR().Height()
+	if err != nil {
+		return false, errors.Wrap(err, "failed to get StateReader height")
+	}
+	rp := rolldpos.MustGetProtocol(protocol.MustGetRegistry(ctx))
+	currentEpochNum := rp.GetEpochNum(srHeight)
+	return csr.IsActiveCandidateAt(cand, rp.GetEpochLastBlockHeight(currentEpochNum+1))
```

Comment:
why should be `currentEpochNum+1`?

Reply:
refer to https://github.com/iotexproject/iotex-core/pull/4062#discussion_r1461829192